File: python/3090.maximum-length-substring-with-two-occurrences.py
class Solution:
    def maximumLengthSubstring(self, s: str) -> int:
        base = ord("a")
        k = 2

        win = [0] * 26
        j = 0
        largest = 0

        for i, ch in enumerate(s):
            win[ord(ch) - base] += 1

            while win[ord(ch) - base] > k:
                win[ord(s[j]) - base] -= 1;
                j += 1

            largest = max(largest, i - j + 1)

        return largest

File: golang/3090.maximum-length-substring-with-two-occurrences.go
func maximumLengthSubstring(s string) int {
	const (
		k    = 2
		base = byte('a')
	)

	win := [26]int{}
	largest := 0
	n := len(s)

	for i, j := 0, 0; i < n; i++ {
		win[s[i]-base]++

		for win[s[i]-base] > k {
			win[s[j]-base]--
			j++
		}

		largest = max(largest, i-j+1)
	}

	return largest
}